components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/util/MetricUtils.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.observability.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.config.mapper.ConfigParser;
+import org.wso2.micro.integrator.observability.metric.handler.MetricReporter;
+import org.wso2.micro.integrator.observability.metric.handler.prometheus.reporter.PrometheusReporter;
+
+import java.util.Map;
+
+public class MetricUtils {
+
+    private static Log log = LogFactory.getLog(MetricUtils.class);
+    private static final String METRIC_REPORTER = "metric_reporter";
+
+    private static MetricReporter metricReporter = null;
+
+    public static MetricReporter getMetricReporter() {
```

Comment:
This is not thread safe. If this is being used by multiple places, then you need to handle it

Reply:
@tishan89, this is called at the server startup by the axis2 handler and the synapse handler. Won't be called by multiple threads at the same time. Further, that won;t modify anything.